src/main/java/org/toradocu/extractor/Comment.java

```diff
@@ -63,56 +70,74 @@ private void decodeHTML() {
    * @param text text of the comment.
    * @param wordsMarkedAsCode blocks of text wrapped in {@literal @code} or {@literal <code></code>}
    */
-  public Comment(String text, List<String> wordsMarkedAsCode) {
-    this.text = text.replaceAll("\\s+", " ");
-    this.wordsMarkedAsCode = wordsMarkedAsCode;
+  public Comment(String text, Map<String, List<Integer>> wordsMarkedAsCode) {
+    this(text);
+    this.wordsMarkedAsCode.putAll(wordsMarkedAsCode);
   }
 
   public String getText() {
     return text;
   }
 
-  public List<String> getWordsMarkedAsCode() {
+  public Map<String, List<Integer>> getWordsMarkedAsCode() {
     return wordsMarkedAsCode;
   }
 
-  @Override
-  public int hashCode() {
-    return Objects.hash(text, wordsMarkedAsCode);
-  }
-
-  @Override
-  public boolean equals(Object obj) {
-    if (this == obj) {
-      return true;
-    }
-    if (!(obj instanceof Comment)) {
-      return false;
-    }
-    Comment that = (Comment) obj;
-    return text.equals(that.text) && wordsMarkedAsCode.equals(that.wordsMarkedAsCode);
-  }
-
   /**
    * Adds to {@link #wordsMarkedAsCode} any words in {@code text} that are marked with the given
    * {@code codePattern}.
    *
-   * @param text text in which look for words marked as code
    * @param codePattern regular expression used to identify the words marked as code
    */
-  private void identifyCodeWords(String text, String codePattern) {
-    Matcher matcher = Pattern.compile(codePattern).matcher(text);
-    while (matcher.find()) {
-      String taggedSubstring = matcher.group(1).trim();
-      String[] words = taggedSubstring.split("\\s+");
-      for (String word : words) {
-        if (!word.isEmpty()) {
-          wordsMarkedAsCode.add(word);
+  private void identifyCodeWords(String codePattern) {
+    String[] subSentences = text.split("\\.");
+    for (String subSentence : subSentences) {
```

Comment:
Because we need codewords during the POS-tagging phase, which works sentence by sentence. About the dot you are right: better specify a space right after.